src/lib/list/list-option.html

```diff
@@ -1,10 +1,17 @@
-<div class="mat-list-item-content"  [class.mat-list-item-content-reverse]="checkboxPosition == 'after'" [class.mat-list-item-disabled]="disabled">
+<div class="mat-list-item-content"
+  [class.mat-list-item-content-reverse]="checkboxPosition == 'after'"
+  [class.mat-list-item-disabled]="disabled"
+  [ngSwitch]="_isCompatibilityMode">
+
   <div class="mat-list-item-ripple" md-ripple
        [mdRippleTrigger]="_getHostElement()"
-       [mdRippleDisabled]="_isRippleDisabled()">
-  </div>
+       [mdRippleDisabled]="_isRippleDisabled()"></div>
+
+  <mat-pseudo-checkbox #autocheckbox *ngSwitchCase="true"
```

Comment:
Andrew and I realized a simpler way of doing this: the component itself can just provide `MATERIAL_COMPATIBILITY_MODE` to false
(which is fine because this component does have any real conflicts)